rlkit/torch/torch_rl_algorithm.py

```diff
@@ -46,187 +46,130 @@ def get_batch(self, idx=None):
-            num_evals = self.num_evals
-        else:
-            num_evals = 1
-
-        paths = []
-        for _ in range(num_evals):
-            paths += self.obtain_eval_paths(idx, eval_task=eval_task, deterministic=True)
+        paths = self.obtain_eval_paths(idx, deterministic=False)
         goal = self.env._goal
         for path in paths:
             path['goal'] = goal # goal
 
         # save the paths for visualization, only useful for point mass
         if self.dump_eval_paths:
-            split = 'test' if eval_task else 'train'
-            logger.save_extra_data(paths, path='eval_trajectories/{}-task{}-epoch{}'.format(split, idx, epoch))
+            logger.save_extra_data(paths, path='eval_trajectories/task{}-epoch{}-run{}'.format(idx, epoch, run))
+
         return paths
 
-    def log_statistics(self, paths, split=''):
-        self.eval_statistics.update(eval_util.get_generic_path_information(
-            paths, stat_prefix="{}_task{}".format(split, self.task_idx),
-        ))
-        # TODO(KR) what are these?
-        self.eval_statistics.update(eval_util.get_generic_path_information(
-            self._exploration_paths, stat_prefix="Exploration_task{}".format(self.task_idx),
-        )) # something is wrong with these exploration paths i'm pretty sure...
-        average_returns = eval_util.get_average_returns(paths)
-        self.eval_statistics['AverageReturn_{}_task{}'.format(split, self.task_idx)] = average_returns
-        goal = self.env._goal
-        dprint('GoalPosition_{}_task'.format(split))
-        dprint(goal)
-        self.eval_statistics['GoalPosition_{}_task{}'.format(split, self.task_idx)] = goal
+    def _do_eval(self, indices, epoch):
+        final_returns = []
+        online_returns = []
+        for idx in indices:
+            runs, all_rets = [], []
+            for r in range(self.num_evals):
+                paths = self.collect_paths(idx, epoch, r)
+                all_rets.append([eval_util.get_average_returns([p]) for p in paths])
+                runs.append(paths)
+            all_rets = np.mean(np.stack(all_rets), axis=0) # avg return per nth rollout
+            final_returns.append(all_rets[-1])
+            online_returns.append(all_rets)
+        return final_returns, online_returns
 
     def evaluate(self, epoch):
         statistics = OrderedDict()
         statistics.update(self.eval_statistics)
         self.eval_statistics = statistics
 
-        ### train tasks
-        dprint('evaluating on {} train tasks'.format(len(self.train_tasks)))
-        train_avg_returns = []
-        for idx in self.train_tasks:
-            dprint('task {} encoder RB size'.format(idx), self.enc_replay_buffer.task_buffers[idx]._size)
-            paths = self.collect_paths(idx, epoch, eval_task=False)
-            train_avg_returns.append(eval_util.get_average_returns(paths))
+        ### sample trajectories from prior for vis
+        prior_paths = []
+        for _ in range(10):
```

Comment:
make 10 a parameter?

Reply:
or better yet I think this should go into `debug` mode since aside from visualization it's just a waste of computation

Reply:
see `258b316`...it's hard-coded to collecting 100 trajs, but that seems ok to me

Reply:
ya seems fine if its just for debugging